sources/code/main/modules/keybind-socket.ts

```diff
@@ -0,0 +1,81 @@
+import net from "net";
+import fs from "fs";
+import keycode from "keycode";
+import { app, ipcMain } from "electron/main";
+
+interface KeybindState {
+  _state?: Record<string, {
+    id: string;
+    action: string;
+    shortcut: [0, number, 4][];
+    managed: boolean;
+    context: string;
+  }>;
+}
+
+export default function bindKeybindSocket(window: Electron.BrowserWindow) {
+  if (!process.env["XDG_RUNTIME_DIR"]) {
+    console.error("[Keybinds] $XDG_RUNTIME_DIR is not defined. Will not bind socket.");
+    return;
+  }
+
```

Comment:
There should be a platform check, obviously we don't want to open UNIX socket on Windows and maybe even on the macOS. Maybe a check for Wayland flags should also be done (i.e. `--ozone-platform`) – I don't think X11 users will benefit from the socket if WebCord could just register the keybind on its own.

Reply:
Oversight on my part. Have added a check for `--ozone-platform=wayland`.

Reply:
Also need to add a check for `--ozone-platform-hint=auto`.